sdks/python/apache_beam/utils/profiler.py

```diff
@@ -113,7 +116,10 @@ def __exit__(self, *args):
           h = self.hpy.heap()
           heap_dump_data = '%s\n%s' % (h, h.more)
           self._upload_profile_data(
-              'memory_profile', heap_dump_data, write_binary=False)
+              self.profile_location,
+              'memory_profile',
+              heap_dump_data,
+              write_binary=False)
 
```

Comment:
I decided to change this to pass in `self.profile_location` because in this context, where it's called, we know that it's not `None`.  Let's us avoid calling `assert` inside `self._upload_profile_data`